workflow/util/merge.go

```diff
@@ -37,3 +39,52 @@ func MergeTo(patch, target *wfv1.Workflow) error {
 	}
 	return nil
 }
+
+// MergeMap will merge all element from right map to left map if it is not present in left.
+func MergeMap(left, right map[string]string) {
+	for key, val := range right {
+		if _, ok := left[key]; !ok {
+			fmt.Println(key, val)
+			left[key] = val
+		}
+	}
+}
+
+// MergeWfSpece will merge the workflow space with order of precedence.
+// 1. Workflow Spec, 2 WorkflowTemplate Spec (WorkflowTemplateRef), 3. WorkflowDefault Spec.
+func MergeWfSpecs(wfSpec, wftSpec, wfDefaultSpec *wfv1.WorkflowSpec) (*wfv1.Workflow, error) {
+	if wfSpec == nil {
+		return nil, fmt.Errorf("invalid Workflow spec")
+	}
```

Comment:
If `wfSpec` is `nil`, then we should just return `nil` instead of an error and let the caller handle it.